### PR TITLE
Fixed handling of missing model

### PR DESCRIPTION
**Pull request by PrestonMclean**
added a try block to catch keyError when an android device has no model.
[Fixes issue #820](https://github.com/beeware/briefcase/issues/820#issue-1335035232)

## PR Checklist:
- [ ] All new features have been tested
- [ ] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

File: changes/820.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Fixed the handling of missing model
```
Comment (freakboy3742) [inline changes/820.bugfix:1]: For consistency, this filename should be `820.bugfix.rst`

File: tests/integrations/android_sdk/AndroidSDK/devices/no_model.out [hunk 2]
```diff
@@ -0,0 +1,2 @@
+List of devices attached
+emulator-5554          device product:sdk_phone_x86 device:generic_x86 transport_id:4
```
Comment (freakboy3742) [inline tests/integrations/android_sdk/AndroidSDK/devices/no_model.out:2]: This file is missing a line ending; the real command output ends with a newline

**Review by freakboy3742 — CHANGES_REQUESTED** [review]
Looks good - a couple of cosmetic fixes, and a merge problem (looks like you forked this branch a couple of days ago before we landed #801, which was a big refactor of how we manage tools). 

One other note: you've submitted this pull request from your `main` branch. Ordinarily, if a PR was this close to being ready, I would make the final tweaks myself; however, because you've submitted from your `main` branch rather than a feature branch in your own fork, I can't make modifications to your pull request (your main branch is always protected from outside contributions).

It's generally a good idea to submit pull requests from a branch on your repository, rather than from you `main`; your `main` should usually be a direct copy of the upstream repository. This will save you a lot of headaches later when you come to do future pull requests (or, if you ever want to have multiple pull requests in flight at the same time).

**Comment by PrestonMclean** [comment]
I made the changes, but I am having a problem with trailing white space for the pre-commit in test_devices.py.

File: changes/820.bugfix.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Fixed the handling of missing model
```
Comment (freakboy3742) [inline changes/820.bugfix.rst:1]: This text is used as a release note, so it should be written in the past tense, in a way that anyone can understand it without context. Something like:

Android emulators that don't provide a model identifier can now be used to launch apps.

**Review by freakboy3742 — CHANGES_REQUESTED** [review]
One small tweak for the release note, and we should be good to go.

Edit: Well, that and fixing the broken test :-)

**Review by freakboy3742 — APPROVED** [review]
Looks great - Thanks for the PR!

**Comment by freakboy3742** [comment]
@PrestonMclean Not sure if that was an accident, but FYI - you don't need to do anything else on this PR; as soon as CI completes, I'll merge it, and it will be closed (as will the ticket for the bug it resolves).

**Comment by PrestonMclean** [comment]
ok

**Pull request merged**